fe/fe-core/src/main/java/org/apache/doris/planner/DistributedPlanner.java

```diff
@@ -863,6 +863,14 @@ private PlanFragment createMergeAggregationFragment(AggregationNode node, PlanFr
             //     childFragment.addPlanRoot(node);
             //     return childFragment;
             // }
+
+            PlanNode childPlan = childFragment.getPlanRoot();
+            if (childPlan instanceof OlapScanNode &&
+                    ((OlapScanNode)childPlan).getOlapTable().satisfyHashDistribution(partitionExprs)) {
```

Comment:
I think we move this check to `createAggregationFragment` 780 line is better?  Because `createMergeAggregationFragment ` should create a merge aggregate fragment.